js/src/dom/manipulator.js

```diff
@@ -43,19 +43,15 @@ const Manipulator = {
       return {}
     }
 
-    const attributes = {
-      ...element.dataset
-    }
-
-    Object.keys(attributes)
+    return Object.keys(element.dataset)
       .filter(key => key.startsWith('bs'))
-      .forEach(key => {
+      // eslint-disable-next-line unicorn/no-reduce
+      .reduce((attributes, key) => {
```

Comment:
How much worse would this be in terms of readability if you used a for or forEach loop? We don't use reduce anywhere else and while I prefer readability, reduce is a special case.

Reply:
Okay understood that's why you are not using `reduce`, I was thinking about it already. Let me update it to `forEach`.

Reply:
Thanks! Just thinking out loud, we might go with your original solution anyway :)

Reply:
Updated 👍

Reply:
Thanks! I say we go with this solution and when we switch the whole data file to use a `Map` we can proceed with any further cleanup. Patches welcome as usual, after we land this :)